sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -419,6 +420,72 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] with PredicateHelper wit
       }
   }
 
+  private def pushDownOffset(
```

Comment:
let's have a single `pushDownLimitAndOffset`. It should be simpler to the planner rule, and we need to match different kind of query plans: LIMIT only, OFFSET only, LIMIT + OFFSET, OFFSET + LIMIT, SORT + LIMT, SORT + LIMIT + OFFSET, etc.